### PR TITLE
Google Maps 9.1.1

### LINKED ISSUE #31: [QUESTION] Google maps API
Have you created the Google maps nugets, or we just need to create them by ourselves? 

If second, then what if I create the nugets for google maps api and put the prefix of your name like it is for firebase? like AdamE.Firebase.iOS.ABTesting but for maps AdamE.Google.iOS.Maps to consolidate the full repo namings?

### PR DESCRIPTION
**Pull request by Digifais**
This PR aims to bump GoogleMaps to v9.1.1, this should close #31.

Note: there have been a few minor automatic formatting changes by the IDE. 🤷

### CODE REVIEW

**Comment by Digifais** [comment]
@AdamEssenmacher Fixed merge conflicts.

**Comment by AdamEssenmacher** [comment]
Perfect, thanks! I'll get to reviewing the remaining parts of the PR shortly.

File: source/Google/Maps/ApiDefinition.cs [hunk 1]
```diff
@@ -69,6 +66,9 @@ interface Constants
 		[Field ("kGMSLayerPanoramaZoomKey", "__Internal")]
 		NSString LayerPanoramaZoomKey { get; }
 
+		[Field ("kGMSLayerPanoramaFOVKey", "__Internal")]
+		NSString LayerPanoramaFOVKey { get; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:70]: No need to move this

File: source/Google/Maps/ApiDefinition.cs [hunk 2]
```diff
@@ -78,23 +78,62 @@ interface Constants
...
 
+		[Field ("kGMSMarkerLayerOpacity", "__Internal")]
+		NSString MarkerLayerOpacity { get; }
+
 		[Field ("kGMSAccessibilityCompass", "__Internal")]
 		NSString AccessibilityCompass { get; }
 
 		[Field ("kGMSAccessibilityMyLocation", "__Internal")]
-		NSString AccessibilityMyLocation { get; }
+		NSString AccessibilityOutOfQuota { get; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:88]: copy-paste error?

File: source/Google/Maps/ApiDefinition.cs [hunk 3]
```diff
@@ -562,6 +600,34 @@ interface MapViewDelegate
...
+		[Export ("frame", ArgumentSemantic.Assign)]
+		CGRect Frame { get; set; }
+
+		// @property (nonatomic) GMSCameraPosition * _Nullable camera;
+		[NullAllowed, Export ("camera", ArgumentSemantic.Assign)]
+		CameraPosition Camera { get; set; }
+
+		// @property (nonatomic) GMSMapID * _Nullable mapID;
+		[NullAllowed, Export ("mapID", ArgumentSemantic.Assign)]
+		MapID MapID { get; set; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:626]: Please rename to `MapId MapId` do be in line with standard .NET naming conventions

File: source/Google/Maps/ApiDefinition.cs [hunk 4]
```diff
@@ -1288,15 +1379,16 @@ interface MapServices
...
+		[Obsolete ("Metal is now the default renderer if this API is not invoked. In the future, Metal rendering will be the only available implementation and this API will be a no-op. We encourage customers to remove calls to this method from their code.")]
 		[Static]
 		[Export ("setMetalRendererEnabled:")]
 		bool SetMetalRendererEnabled (bool enabled);
 
 		// + (void)setAbnormalTerminationReportingEnabled:(BOOL)enabled;
 		[Static]
 		[Export ("setAbnormalTerminationReportingEnabled:")]
-		bool AbnormalTerminationReportingEnabled (bool enabled);
+		bool SetAbnormalTerminationReportingEnabled (bool enabled);
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1391]: No need to rename this

File: source/Google/Maps/ApiDefinition.cs [hunk 5]
```diff
@@ -1488,12 +1585,214 @@ interface TextureStyle {
...
+	[BaseType (typeof (NSObject), Name = "GMSPlaceFeature")]
+	[DisableDefaultCtor]
+	interface PlaceFeature : Feature {
+		// @property (readonly, nonatomic) GMSFeatureType _Nonnull featureType;
+		[Export ("featureType")]
+		string FeatureType { get; }
+
+		// @property (readonly, nonatomic) NSString * _Nonnull placeID;
+		[Export ("placeID")]
+		string PlaceID { get; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1726]: PlaceId

File: source/Google/Maps/ApiDefinition.cs [hunk 6]
```diff
@@ -646,14 +714,42 @@ interface MapView
...
+		[Obsolete ("Use constructor with MapViewOptions instead.")]
 		[Static]
 		[Export ("mapWithFrame:camera:")]
 		MapView FromCamera (CGRect frame, CameraPosition camera);
 
+		// +(instancetype _Nonnull)mapWithFrame:(CGRect)frame mapID:(GMSMapID * _Nonnull)mapID camera:(GMSCameraPosition * _Nonnull)camera __attribute__((availability(swift, unavailable))) __attribute__((deprecated("Use -init or -initWithOptions: instead.")));
+		[Obsolete ("Use constructor with MapViewOptions instead.")]
+		[Static]
+		[Export ("mapWithFrame:mapID:camera:")]
+		MapView FromCamera (CGRect frame, MapID mapID, CameraPosition camera);
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:741]: mapId

File: source/Google/Maps/ApiDefinition.cs [hunk 7]
```diff
@@ -646,14 +714,42 @@ interface MapView
...
+
+		// -(instancetype _Nonnull)initWithFrame:(CGRect)frame camera:(GMSCameraPosition * _Nonnull)camera __attribute__((deprecated("Use -init or -initWithOptions: instead.")));
+		[Obsolete ("Use constructor with MapViewOptions instead.")]
+		[Export ("initWithFrame:camera:")]
+		NativeHandle Constructor (CGRect frame, CameraPosition camera);
+
+		// -(instancetype _Nonnull)initWithFrame:(CGRect)frame mapID:(GMSMapID * _Nonnull)mapID camera:(GMSCameraPosition * _Nonnull)camera __attribute__((deprecated("Use -init or -initWithOptions: instead.")));
+		[Obsolete ("Use constructor with MapViewOptions instead.")]
+		[Export ("initWithFrame:mapID:camera:")]
+		NativeHandle Constructor (CGRect frame, MapID mapID, CameraPosition camera);
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:751]: mapId

File: source/Google/Maps/ApiDefinition.cs [hunk 8]
```diff
@@ -679,18 +775,13 @@ interface MapView
...
+		// -(GMSFeatureLayer<GMSPlaceFeature *> * _Nonnull)featureLayerOfFeatureType:(GMSFeatureType _Nonnull)featureType __attribute__((swift_name("featureLayer(of:)")));
+		[Export ("featureLayerOfFeatureType:")]
+		FeatureLayer FeatureLayerOf (string featureType);
 
-		// -(instancetype _Nonnull)initWithFrame:(CGRect)frame mapID:(GMSMapID * _Nonnull)mapID camera:(GMSCameraPosition * _Nonnull)camera;
-		[Export ("initWithFrame:mapID:camera:")]
-		NativeHandle Constructor (CGRect frame, MapId mapId, CameraPosition camera);
+		// -(GMSDatasetFeatureLayer * _Nonnull)datasetFeatureLayerOfDatasetID:(NSString * _Nonnull)datasetID __attribute__((swift_name("datasetFeatureLayer(of:)")));
+		[Export ("datasetFeatureLayerOfDatasetID:")]
+		DatasetFeatureLayer DatasetFeatureLayerOf (string datasetID);
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:784]: datasetId

File: source/Google/Maps/ApiDefinition.cs [hunk 9]
```diff
@@ -1488,12 +1585,214 @@ interface TextureStyle {
...
+		string FeatureType { get; }
+	}
+
+	// @interface GMSDatasetFeature : NSObject <GMSFeature>
+	[BaseType (typeof (NSObject), Name = "GMSDatasetFeature")]
+	[DisableDefaultCtor]
+	interface DatasetFeature : Feature {
+		// @property (readonly, nonatomic) NSString * _Nonnull datasetID;
+		[Export ("datasetID")]
+		string DatasetID { get; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1626]: DatasetId

File: source/Google/Maps/ApiDefinition.cs [hunk 10]
```diff
@@ -1488,12 +1585,214 @@ interface TextureStyle {
...
+
+		// @property (readonly, nonatomic) NSDictionary<NSString *,NSString *> * _Nonnull datasetAttributes;
+		[Export ("datasetAttributes")]
+		NSDictionary<NSString, NSString> DatasetAttributes { get; }
+
+		// -(instancetype _Nonnull)initWithDatasetID:(NSString * _Nonnull)datasetID datasetAttributes:(NSDictionary<NSString *,NSString *> * _Nonnull)datasetAttributes __attribute__((objc_designated_initializer));
+		[Export ("initWithDatasetID:datasetAttributes:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (string datasetID, NSDictionary<NSString, NSString> datasetAttributes);
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1635]: datasetId

File: source/Google/Maps/ApiDefinition.cs [hunk 11]
```diff
@@ -1488,12 +1585,214 @@ interface TextureStyle {
...
+		[Export ("initWithFeatureType:")]
+		NativeHandle Constructor (string featureType);
+	}
+
+	// @interface GMSDatasetFeatureLayer : GMSFeatureLayer
+	[BaseType (typeof (FeatureLayer), Name = "GMSDatasetFeatureLayer")]
+	interface DatasetFeatureLayer {
+		// @property (readonly, nonatomic) NSString * datasetID;
+		[Export ("datasetID")]
+		string DatasetID { get; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1664]: DatasetId

File: source/Google/Maps/ApiDefinition.cs [hunk 12]
```diff
@@ -1488,12 +1585,214 @@ interface TextureStyle {
...
+		[Export ("featureType")]
+		string FeatureType { get; }
+
+		// @property (readonly, nonatomic) NSString * _Nonnull placeID;
+		[Export ("placeID")]
+		string PlaceID { get; }
+
+		// -(instancetype _Nonnull)initWithFeatureType:(GMSFeatureType _Nonnull)featureType placeID:(NSString * _Nonnull)placeID;
+		[Export ("initWithFeatureType:placeID:")]
+		NativeHandle Constructor (string featureType, string placeID);
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1730]: placeId

File: source/Google/Maps/ApiDefinition.cs [hunk 13]
```diff
@@ -1502,8 +1801,11 @@ interface MapId : INSCopying {
...
 		[Export ("mapIDWithIdentifier:")]
-		MapId MapIdWithIdentifier (string identifier);
-	}
+		MapID MapIdWithIdentifier (string identifier);
 
-	#endregion
+		// @property (readonly, nonatomic, class) NS_SWIFT_NAME(demoMapID) GMSMapID * demoMapID __attribute__((swift_name("demoMapID")));
+		[Static]
+		[Export ("demoMapID")]
+		MapID DemoMapID { get; }
```
Comment (AdamEssenmacher) [inline source/Google/Maps/ApiDefinition.cs:1809]: DemoMapId

**Review by AdamEssenmacher — CHANGES_REQUESTED** [review]
I think we're 99% of the way there! I think I only spotted one actual error. The rest of my comments either ask to not move code around unnecessarily, to avoid unnecessarily changing the existing API surface, and to use the .NET naming convention for 'Id'.

I'm really excited to be able to bring this in soon. Great work!

**Comment by Digifais** [comment]
@AdamEssenmacher PR comments should be fixed.

**Comment by AdamEssenmacher** [comment]
Excellent. I checked a test build using the sample app from a port of [Xamarin.Forms.GoogleMaps](https://github.com/amay077/Xamarin.Forms.GoogleMaps) and everything looked good!

If you'd be interested in joining this project as a contributor, I'll send you and invite. For a contribution like this, I'd be happy to add you to the FUNDING.yml too :)

**Review by AdamEssenmacher — APPROVED** [review]
LGTM

**Comment by Digifais** [comment]
> Excellent. I checked a test build using the sample app from a port of [Xamarin.Forms.GoogleMaps](https://github.com/amay077/Xamarin.Forms.GoogleMaps) and everything looked good!
> 
> If you'd be interested in joining this project as a contributor, I'll send you and invite. For a contribution like this, I'd be happy to add you to the FUNDING.yml too :)

Adam, I think this would be great opportunity for me to give back to the .NET community next to the packages I already try to maintain once in a while (https://www.nuget.org/profiles/Digifais) 😉. Let's try and put some work in this repo to bring more packages up to date!

**Comment by AdamEssenmacher** [comment]
Awesome! I'll send you an invite shortly. If you want to send me an email at hello@scavos.com maybe we can exchange Discords?

I'm really happy to see those Facebook bindings too. They've been on my list for a long while. Are there public repos for them?

**Pull request merged**